CMakeLists.txt

```diff
@@ -40,6 +40,54 @@ else()
   message(WARNING "Unknown compiler... recklessly proceeding without a version check")
 endif()
 
+set(LIBMONGOC_REQUIRED_VERSION 1.22.1) # TODO: update to 1.24.0 once released.
+set(LIBMONGOC_REQUIRED_ABI_VERSION 1.0)
+
+set(NEED_DOWNLOAD_C_DRIVER false)
+if(TARGET mongoc_shared OR TARGET mongoc_static)
+  # If these targets exist, then libmongoc has already been included as a project
+  # sub-directory
+  message (STATUS "Found libmongoc targets declared in current build scope; version not checked")
+
+  if(NOT MONGOCXX_LINK_WITH_STATIC_MONGOC)
+    set(libmongoc_target mongoc_shared)
+  else()
+    set(libmongoc_target mongoc_static)
+  endif()
+
+  if(MONGOCXX_BUILD_STATIC)
+    set(mongocxx_pkg_dep "find_dependency(mongoc-1.0 REQUIRED)")
+  endif()
+else()
+    find_package(mongoc-${LIBMONGOC_REQUIRED_ABI_VERSION} ${LIBMONGOC_REQUIRED_VERSION} QUIET)
+    if(mongoc-${LIBMONGOC_REQUIRED_ABI_VERSION}_FOUND)
+        message(STATUS "Already found libmongoc, don't need to download it.")
+    else()
+        set(NEED_DOWNLOAD_C_DRIVER true CACHE INTERNAL "")
+    endif()
+endif()
+
+if(NEED_DOWNLOAD_C_DRIVER)
+    message(STATUS "No Mongo C Driver path provided via CMAKE_PREFIX_PATH, will download a release from the internet.")
+    include(FetchContent)
+    include(ExternalProject)
+
+    # Declare mongo-c-driver as a dependency
+    FetchContent_Declare(
+      mongo-c-driver
+      GIT_REPOSITORY https://github.com/mongodb/mongo-c-driver.git
+      GIT_TAG b9ca36ff8c64f12330326f080c725c5502294834
+    )
+
+    FetchContent_MakeAvailable(mongo-c-driver)
```

Comment:
The current changes also install the C driver:

```
cmake -S. -Bcmake-build -DCMAKE_CXX_STANDARD=17
cmake --build cmake-build --target install
# Results in installing C driver files. Example:
# /Users/kevin.albertson/review/mongo-cxx-driver-967/cmake-build/install/lib/libmongoc-1.0.0.dylib
```

Consumers of the C++ driver may not need a separate install of the C driver.

Consider disabling install of the C driver, and static linking to the C driver.

Reply:
Sounds good!